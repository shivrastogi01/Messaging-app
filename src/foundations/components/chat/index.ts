export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
