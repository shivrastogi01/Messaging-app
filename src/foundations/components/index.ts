export * from "./presentation";
export * from "./layout";
