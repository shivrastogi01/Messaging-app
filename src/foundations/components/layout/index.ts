export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
