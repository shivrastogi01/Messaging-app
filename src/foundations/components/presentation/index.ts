export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
