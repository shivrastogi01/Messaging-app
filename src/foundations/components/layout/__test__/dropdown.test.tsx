import React from "react";
import renderWithTheme from "main/testing/render-with-theme";
import { fireEvent } from "@testing-library/react";
import { Dropdown } from "foundations/components/layout";
import { Icons } from "foundations/components/presentation";

describe("Picker", () => {
  test("matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Dropdown
        icon={Icons.Add}
        title="show picker"
        render={() => <div></div>}
      ></Dropdown>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("content is hidden by default", () => {
    const { getByText } = renderWithTheme(
      <div>
        <Dropdown
          icon={Icons.Add}
          title="show picker"
          render={() => <p>content</p>}
        ></Dropdown>
      </div>
    );
    // should not be showing the content
    expect(() => {
      getByText("content");
    }).toThrow();
  });

  test("content is shown after clicking icon", () => {
    const { getByText, getByTitle } = renderWithTheme(
      <div>
        <Dropdown
          icon={Icons.Add}
          title="show picker"
          render={() => <p>content</p>}
        ></Dropdown>
      </div>
    );
    const icon = getByTitle("show picker");
    fireEvent.click(icon);
    // should not be showing the content
    expect(getByText("content")).toBeTruthy();
  });

  test("content is hidden after clicking outside", () => {
    const { getByText, getByTitle } = renderWithTheme(
      <div>
        <p>outside</p>
        <Dropdown
          icon={Icons.Add}
          title="show picker"
          render={() => <p>content</p>}
        ></Dropdown>
      </div>
    );
    const outside = getByText("outside");
    const icon = getByTitle("show picker");
    fireEvent.click(icon);
    fireEvent.click(outside);
    // should not be showing the content
    expect(() => {
      getByText("content");
    }).toThrow();
  });

  test("render prop hides content", () => {
    const { getByText, getByTitle } = renderWithTheme(
      <div>
        <Dropdown
          icon={Icons.Add}
          title="show picker"
          render={dismiss => {
            return (
              <button
                onClick={() => {
                  dismiss();
                }}
              >
                content
              </button>
            );
          }}
        ></Dropdown>
      </div>
    );
    const icon = getByTitle("show picker");
    fireEvent.click(icon);
    const childButton = getByText("content");
    fireEvent.click(childButton);
    // should not be showing the content
    expect(() => {
      getByText("content");
    }).toThrow();
  });
});
