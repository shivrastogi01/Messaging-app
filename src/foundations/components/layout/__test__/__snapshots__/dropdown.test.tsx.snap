// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker matches the snapshot 1`] = `
.c1 {
  position: absolute;
  z-index: 200;
}

.c0 {
  cursor: pointer;
}

.c0 svg {
  display: block;
}

<div>
  <object
    class="c0"
  >
    <svg
      title="show picker"
    >
      add.svg
    </svg>
  </object>
  <div
    class="c1"
  />
</div>
`;
